lib/socket.js

```diff
@@ -305,23 +305,23 @@ Socket.prototype.probe = function (name) {
       var upgradeLosesBinary = !this.supportsBinary && self.transport.supportsBinary;
       failed = failed || upgradeLosesBinary;
     }
-    if (failed) return;
+    if (failed) self.emit('upgradeError', err);;
```

Comment:
if they were originally returning here we should as well. After emitting we should immediately return.